rtl/eclrtl/eclhelper_dyn.cpp

```diff
@@ -0,0 +1,268 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "platform.h"
+#include "jptree.hpp"
+#include "eclrtl.hpp"
+#include "eclhelper.hpp"
+#include "rtlds_imp.hpp"
+#include "eclhelper_base.hpp"
+#include "eclhelper_dyn.hpp"
+
+#include "rtlfield.hpp"
+#include "rtlrecord.hpp"
+#include "rtldynfield.hpp"
+#include "rtlkey.hpp"
+
+//---------------------------------------------------------------------------
+static void readString(StringBuffer &out, const char * &in)
+{
+    for (;;)
+    {
+        char c = *in++;
+        if (!c)
+            throw MakeStringException(0, "Invalid filter - missing closing '");
+        if (c=='\'')
+            break;
+        if (c=='\\')
+            UNIMPLEMENTED;
+        out.append(c);
+    }
+}
+class ECLRTL_API CDynamicDiskReadArg : public CThorDiskReadArg
+{
+public:
+    CDynamicDiskReadArg(const char *_fileName, IOutputMetaData *_in, IOutputMetaData *_out, unsigned __int64 _chooseN, unsigned __int64 _skipN, unsigned __int64 _rowLimit)
+        : fileName(_fileName), in(_in), out(_out), chooseN(_chooseN), skipN(_skipN), rowLimit(_rowLimit)
+    {
+        inrec = &in->queryRecordAccessor(true);
+        numOffsets = inrec->getNumVarFields() + 1;
+        translator.setown(createRecordTranslator(queryOutputMeta()->queryRecordAccessor(true), *inrec));
+    }
+    virtual bool needTransform() override
+    {
+        return true;
+        //return translator->needsTranslate(); might be more appropriate?
```

Comment:
should this be different?

Reply:
True will work fine for now - there may be a point in the future where we could avoid it if code is refactored elsewhere, but that should be the subject of a different Jira I think.